lib/sync-DataSetModel.js

```diff
@@ -371,33 +385,42 @@ var self = {
 
       var hashes = [];
       var recOut = {};
-      for (var i in records) {
-        var rec = {};
-        var recData = records[i];
-        var hash = syncUtil.generateHash(recData);
-        hashes.push(hash);
-        rec.data = recData;
-        rec.hash = hash;
-        recOut[i] = rec;
-      }
-      var globalHash = syncUtil.generateHash(hashes);
 
-      var previousHash = datasetClient.dataHash ? datasetClient.dataHash : '<undefined>';
-      syncUtil.doLog(dataset.id, 'verbose', 'doSyncList cb ' + ( cb !== undefined) + ' - Global Hash (prev :: cur) = ' + previousHash + ' ::  ' + globalHash);
+      async.forEachSeries(Object.keys(records), function(recordId, itemCallback){
+        var record = records[recordId];
+        syncUtil.asyncGenerateHash(record, function(err, recordHash) {
+          var rec = {
+            data: record,
+            hash: recordHash
+          };
+          hashes.push(recordHash);
+          recOut[recordId] = rec;
+          itemCallback();
+        });
+      }, function(err) {
+        if (err) return cb(err);
 
-      datasetClient.dataHash = globalHash;
-      self.datasetClientRecords[datasetClient.id] = recOut;
+        syncUtil.asyncGenerateHash(hashes, function(err, globalHash) {
+          console.log('doListHandler callback - Async : ' + globalHash);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this logging statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS2228)